web/yo/app/scripts/directives/searchadddrug.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+angular.module('oncokbApp')
+    .directive('searchAddDrug', function ($rootScope, DatabaseConnector, dialogs, _, mainUtils, $q, FirebaseModel) {
+        return {
+            templateUrl: 'views/searchAddDrug.html',
+            restrict: 'AE',
+            controller: function ($scope) {
+                function checkSameDrug(drugName, code) {
+                    var isSame = false;
+                    var drugKeys = mainUtils.getKeysofArray($scope.drugList);
+                    _.find(drugKeys, function (key) {
+                        if (((code === '') && (drugName === $scope.drugList[key].drugName)) || ((code !== '') && (code === $scope.drugList[key].ncitCode))) {
```

Comment:
When there is a code, ((code !== '') && (code === $scope.drugList[key].ncitCode))), code instead of the name will be checked. But when the code is blank, names will be checked.